qadence/backends/pulser/backend.py

```diff
@@ -350,10 +353,18 @@ def expectation(
                 # TODO: There should be a better check for batched density matrices.
                 if dms.size()[0] > 1:
                     res_list = [
-                        [obs.native(dm, param_values, qubit_support=support) for dm in dms]
+                        [
+                            obs.native(
+                                dm.squeeze(), param_values, qubit_support=support, noise=noise
```

Comment:
so we assume dim `0` is always 1? if not, that can result in undefined behaviour

Reply:
Yes. At this point, `dms` are batched density matrices by the noise parameters. So `dm` is a single density matrix.

Reply:
ah ok